custom/logistics/api.py

```diff
@@ -255,5 +257,12 @@ def add_language_to_user(self, logistics_sms_user):
                 v = VerifiedNumber.by_phone(phone_number, include_pending=True)
                 if v:
                     v.delete()
-                user.save_verified_number(self.domain, phone_number, True,
-                                          logistics_sms_user.backend)
+                try:
+                    user.save_verified_number(self.domain, phone_number, True,
+                                              logistics_sms_user.backend)
+                except PhoneNumberInUseException:
+                    v = VerifiedNumber.by_phone(phone_number, include_pending=True)
+                    if v.domain in domains:
+                        v.delete()
+                        user.save_verified_number(self.domain, phone_number, True,
+                                                  logistics_sms_user.backend)
```

Comment:
@kkrampa @czue Note that when calling save_verified_number, the backend_id is optional, and it's usually recommended to leave it blank so that the system will either choose the project's default gateway or the best global gateway for the destination number. (I've made the documentation clearer in #5799)